dvc/exceptions.py

```diff
@@ -31,24 +31,25 @@ class OutputDuplicationError(DvcException):
         stages (list): list of paths to stages.
     """
 
-    def __init__(self, output, stages):
+    def __init__(self, output: str, stages: List[Any]):
```

Comment:
I would prefer this to be a set. Need to import `Set` from typing, and import `Stage` inside a `TYPE_CHECKING` block.

```suggestion
    def __init__(self, output: str, stages: Set["Stage"]) -> None:
```